### PR TITLE
36 evaluator fix impl

### PR DESCRIPTION
**Pull request by PawelDabrowski83**
As Evaluator operates on Gene mock I was wondering how to check if value is set correctly. I have used ArgumentCaptor to verify if fitness is calculated properly. This is my first usage of this class, so any feedback is welcome.

### CODE REVIEW

**Review by KamilKamilS — COMMENTED** [review]
I am wondering if we should stay with to asserts in tests, as in the comment below

File: src/test/java/com/facebook/genAlgo/evaluator/Variant1EvaluatorImplTest.java [hunk 1]
```diff
@@ -30,9 +32,10 @@ void setFitnessFullMatch(float fitnessExpected, char target, char evaluatedValue
 
         // when
         evaluator.setFitness(gene);
+        verify(gene).setFitness(floatArgumentCaptor.capture());
 
         // then
-        assertEquals(fitnessExpected, gene.getFitness(), 0.0001);
+        assertEquals(fitnessExpected, floatArgumentCaptor.getValue(), 0.0001);
```
[the remark this answers is not in the document]
Comment (PawelDabrowski83) [reply src/test/java/com/facebook/genAlgo/evaluator/Variant1EvaluatorImplTest.java:38]: gene is a mock and it mocks all of Gene.class methods so we cannot check gene.getFitness() now.

But I think I have an idea how to write another test to be sure that fitness is set properly.

[the remark this answers is not in the document]
Comment (KamilKamilS) [reply src/test/java/com/facebook/genAlgo/evaluator/Variant1EvaluatorImplTest.java:38]: Yes you have right. So maybe in GeneTest we should check correctness of setFitness() method?

[the remark this answers is not in the document]
Comment (PawelDabrowski83) [reply src/test/java/com/facebook/genAlgo/evaluator/Variant1EvaluatorImplTest.java:38]: Well, I am testing Evaluator in this class, so I assume that Gene.class works perfect and its tested separately.

[the remark this answers is not in the document]
Comment (PawelDabrowski83) [reply src/test/java/com/facebook/genAlgo/evaluator/Variant1EvaluatorImplTest.java:38]: As a different idea I thought that I could mock RandomProvider and work on regular Gene object with all its methods. This does not require ArgumentCaptor, that is a pity, but I think it is a simpler solution to read.

**Review by KamilKamilS — APPROVED** [review]

**Pull request merged**